pkg/clients/clients.go

```diff
@@ -42,13 +46,62 @@ import (
 
 const SystemNamespace = "cattle-system"
 
+type ClientInterface interface {
+	EventRecorder(name string) record.EventRecorder
+	Start(ctx context.Context) error
+	Elemental() elmcontrollers.Interface
+	Core() corecontrollers.Interface
+	RBAC() rbaccontrollers.Interface
+	K8s() kubernetes.Interface
+	CAPI() capi.Interface
+	Apply() apply.Apply
+	Fleet() fleetcontrollers.Interface
+	Rancher() ranchercontrollers.Interface
+}
+
 type Clients struct {
-	*clients.Clients
-	Fleet     fleetcontrollers.Interface
-	Elemental elmcontrollers.Interface
-	Events    corev1Typed.EventInterface
-	Rancher   ranchercontrollers.Interface
-	CAPI      capi.Interface
+	sharedControllerFactory controller.SharedControllerFactory
```

Comment:
This drops the generic clients.Clients to include only the needed clients.

I dont event think the `Events` interface is used